### PR TITLE
Improve touch-behaviour

### PR DESCRIPTION
**Pull request by dennisroethig**
### Version
`4.1.0` -> `5.0.0`

### Overview
I noticed that there are a few issues with the tooltips on iOS devices and the interpretation of the mouse and click events. This is a first draft of adjusting the existing events to make it work across devices

##### Existing behaviour
- iOS devices pick up the tap as `mouseenter` on the first tap. After that the `click` event is triggered which causes the tooltip to close (and therefore not to re-open if you want it). The workaround for this is to tap somewhere else and then the tap will be registered again as `mouseenter`. I'm not sure if that broke recently or was just never picked up.

- Using the tooltips on anchors causes another issue on touch devices. iOS requires the user to tap twice to follow the link (first tap shows tooltip), Android follows the link after the first tap.

##### Changed behaviour
- Using a mouse will show the tooltip on `mouseenter` and hides it on `mouseleave` as before. Clicking on the tooltip-trigger is not closing the tooltip anymore (I can't see why it would be a requirement...)

- Using a touch-device, the `touchstart` event will trigger a toggle between show/hide

### Concerns
I tested this "fix" on different browsers and my iPhone 6 and Nexus 7. As far as I can see this is the most streamlined behaviour. Below are the usages for our apps and as it's changing the behaviour I would prefer to make a major version bump and get the apps updating whenever they're fine with the change.

##### Existing usage
[Contests](https://github.com/99designs/contests/search?utf8=%E2%9C%93&q=data-tooltip)
[Handover](https://github.com/99designs/handover/search?utf8=%E2%9C%93&q=data-tooltip)
[Tasks](https://github.com/99designs/tasks/search?utf8=%E2%9C%93&q=data-tooltip)
[LogoStore](https://github.com/99designs/logo_store/search?utf8=%E2%9C%93&q=data-tooltip&type=Code)

##### Not using asimov-tooltip
[Workbench](https://github.com/99designs/workbench/search?utf8=%E2%9C%93&q=data-tooltip&type=Code)
[Projects](https://github.com/99designs/projects/search?utf8=%E2%9C%93&q=data-tooltip&type=Code)
[Start](https://github.com/99designs/start/search?utf8=%E2%9C%93&q=data-tooltip&type=Code)

### Todo
- [x] Clarify what the existing `on click` is supposed to do
- [x] Make sure existing tooltips are still working
- [ ] Cross-device testing

### CODE REVIEW

File: lib/tooltip.js [hunk 1]
```diff
@@ -176,11 +176,18 @@
             .on(options.show, function tooltipStart() {
                 that.show($elem, $tooltip, options);
             })
-            .mouseleave(function(e) {
+            .on('mouseleave', function() {
                 that.hide($tooltip);
             })
-            .on('click', function tooltipEnd() {
```
Comment (dennisroethig) [inline lib/tooltip.js]: @joshamos do you have an idea what this `hide on click` is supposed to be used for?

**Comment by dennisroethig** [comment]
Pink @andrewk @dannymidnight @xzyfer @joshamos @BeiYang123 @MatthewDiamant

File: lib/tooltip.js (hunk 1 above)
Comment (joshamos) [inline lib/tooltip.js]: hide upon clicking on the open tooltip?

Comment (dennisroethig) [inline lib/tooltip.js]: yeah but what's the real use-case for that. The tooltip will disappear on mouseleave, so there should be no need for this on your typical desktop machine. And touch-events would just toggle show/hide ... does that make sense or am I missing something? :)

Comment (joshamos) [inline lib/tooltip.js]: Great point, no - I can't think of why this actually exists.

**Comment by dannymidnight** [comment]
@dennisroethig Great write up! Yeah I think sounds reasonable and can't think why you would want to hide a tooltip on click.

**Comment by andrewk** [comment]
+1 to Chris' points above. I especially appreciate the clarity of the PR.

**Comment by dennisroethig** [comment]
Thanks for your time guys :+1:

**Pull request merged**